### PR TITLE
Refactor DAOs and Models

### PR DESCRIPTION
**Pull request by tobiasrohloff**
This PR migrates all database accesses to the DAOs and introduces some conventions for them:
- single entity requests begin with `find`, collection requests begin with `all`
- managed requests with results wrapped as `LiveData` are implemented as instance methods of each DAO
  - some sugar for default `all` and `find` cases is implemented by the `BaseDao`
- unmanaged requests with plain object/list results are available via an `Unmanaged` object in each DAO
  - requests are class methods, since no managed realm instance and therefore no state is needed
  - a new `#asCopy` extensions must be called to ensure no Realm artifacts are used outside the DAOs
- some model classes still have convenient accessor methods to return relationship object
  - these are always unmanaged and only call the corresponding DAO method

In the future, I will try to further enforce some of these conventions, but at least all the code lives inside the DAO classes now and we have a standartized way of accessing managed and unmanaged data.

This PR also fixes https://github.com/openHPI/xikolo-android/issues/109.

### CODE REVIEW

File: app/src/main/java/de/xikolo/controllers/channels/ChannelDetailsActivity.kt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    lateinit var imageView: ImageView
+
+    @BindView(R.id.appbar)
+    lateinit var appBarLayout: AppBarLayout
+
+    @BindView(R.id.collapsing_toolbar)
+    lateinit var collapsingToolbar: CollapsingToolbarLayout
+
+    @BindView(R.id.stub_bottom)
+    lateinit var stubBottom: ViewStub
```
Comment (bengelhaupt) [inline app/src/main/java/de/xikolo/controllers/channels/ChannelDetailsActivity.kt]: This variable is unused, should it be removed?

File: app/src/main/java/de/xikolo/models/dao/CourseDao.kt [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                            EnrollmentDao.Unmanaged.findForCourse(course.id)
+                                ?.anyCertificateAchieved() == true
+                        }
+                }
+
+            fun search(query: String?, withEnrollment: Boolean): List<Course> =
+                Realm.getDefaultInstance().use { realm ->
+                    val realmQuery = realm.where<Course>()
+                        .equalTo("external", false)
+                        .beginGroup()
```
Comment (bengelhaupt) [inline app/src/main/java/de/xikolo/models/dao/CourseDao.kt]: group indentation

File: app/src/main/java/de/xikolo/models/Video.java [hunk 3]
```diff
@@ -44,14 +44,6 @@
...
 
-    public static Video get(String id) {
-        Realm realm = Realm.getDefaultInstance();
-        Video model = realm.where(Video.class).equalTo("id", id).findFirst();
-        if (model != null) model = realm.copyFromRealm(model);
-        realm.close();
-        return model;
-    }
-
     public static Video getForContentId(String contentId) {
```
Comment (bengelhaupt) [inline app/src/main/java/de/xikolo/models/Video.java]: why is this left over?

File: app/src/main/java/de/xikolo/controllers/channels/ChannelDetailsActivity.kt (hunk 1 above)
Comment (tobiasrohloff) [reply app/src/main/java/de/xikolo/controllers/channels/ChannelDetailsActivity.kt]: Done.

File: app/src/main/java/de/xikolo/models/dao/CourseDao.kt (hunk 2 above)
Comment (tobiasrohloff) [reply app/src/main/java/de/xikolo/models/dao/CourseDao.kt]: Fixed.

File: app/src/main/java/de/xikolo/models/Video.java (hunk 3 above)
Comment (tobiasrohloff) [reply app/src/main/java/de/xikolo/models/Video.java]: Removed.

**Comment by tobiasrohloff** [comment]
I added some docs to the `BaseDao`.

**Pull request merged**